Standards/scs-XXXX-v1-security-of-iaas-service-software.md

```diff
@@ -0,0 +1,108 @@
+All releases that are still maintained can be looked up at the releases page from OpenStack[^1].
+
+Allowing maintained versions would give CSPs a little bit more time to update and test their environments, while still receiving relevant security updates and bug fixes.
+Also CSPs that want to become SCS-compliant may not have the burden to upgrade their deployments, but can test before an upgrade, where they need to put in additional work to become SCS-compliant.
+
+One problem is, that there might be new features implemented in the newest versions of the software, which are desired by other standards to be SCS-compliant.
+In that case allowing all maintained versions would lead to a two-year timespan customers would need to wait for before such a feature becomes available in all SCS-compliant deployments.
+In case of security relevant features this is not advisable.
+
+#### Standards implicitly define the minimum versions of Software
+
+Instead of requiring a defined minimum software version, it could be derived from the standards.
+Because: Whenever there is a new wanted behavior a standard should be created and a resonable timeframe given to CSPs to adopt a software version that can fulfill the new standard.
+Through the combination of all standards that are in place, the minimum version for the IaaS service software is implicitly given.
+
+This would avoid to have conflicting versions of software in terms of feature parity, while also allowing older software.
+Using this approach requires an additional advise to CSPs to update or implement patches for security issues.
+
+#### Advise CSPs to integrate software updates
+
+As long as maintained versions of software are used, updates with security patches are available and only need to be integrated.
+This can and should be done in a reasonable short timeframe.
+
+But CSPs may even use releases of IaaS software, that are either not maintained anymore by an open source community or may be even closed source implementations of the mandatory IaaS APIs.
+Allowing older versions or closed source software would only be acceptable, when CSPs assure (e.g. in documentation), that they themself will patch the software within their deployments.
+Security bug fixes must be implemented and proof of the fix then provided.
+Only under these circumstances deployments with older or alternative IaaS Layer software may be handled as compliant.
+
+This option could be taken for granted, but to actually advise using it may encourage CSPs to take a closer look on their life-cycle management and security risk handling.
+And CSPs using OpenStack could even be encouraged to upgrade their deployments.
+
+## Standard for a minimum IaaS Layer Software version
+
+If a deployment is affected by a security issue and a maintained[^1] version of OpenStack is used as implementation for IaaS Layer software, security patches noted in OSSNs and OSSAs MUST be integrated within a reasonable timeframe.
```

Comment:
What is a reasonable time? IMO, standard should be explicit here in order to guide CSPs.

How does customers know, whether or not a specific bug is fixed? Do we "force" CSPs to document software version/patch version somewhere?

Reply:
I can only propose a timeframe, but this need to be discussed with CSPs. I think i will attend the lean operator coffee next monday to ask for their opinion.

Reply:
In the newest version we have timeframes.
We cannot force CSPs to document software versions, but for higher versions than SCS-compliant, like scs-open there is a requirement for SBOMs.
In the end, we must trust the CSPs to update their deployments fast.